src/main/java/org/reactome/addlinks/IdentifiersDotOrgUtil.java

```diff
@@ -69,11 +70,27 @@ public static String getAccessUrlForResource(String resourceIdentifier)
 					case HttpStatus.SC_OK:
 						JsonReader reader = Json.createReader(new StringReader(responseString));
 						JsonObject responseObject = reader.readObject();
-						// Leave the {$id} in the URL and let the caller replace it.
-						accessUrl = responseObject.getString(ACCESS_URL_TOKEN).toString().replaceAll("\"", "");
+						String responseAccessURL = responseObject.getString(ACCESS_URL_TOKEN).toString().replaceAll("\"", "");
+						// Before accepting the AccessURL from the response,
+						// check that the identifier is not deprecated. If identifiers.org
+						// deprecates an identifier, the AccessURL in the response
+						// will look something like this: https://registry.identifiers.org/deprecation/resources/MIR:00100113/{$id}
+						// (this example uses the deactivated/deprecated Rhea identifier that caused a problem
+						// during Releaes 77)
+						// But the HTTP Response code is 200 so we can't just blindly replace our Access URL with the
+						// one from identifiers.org.
+						if (responseAccessURL.contains("registry.identifiers.org/deprecation"))
+						{
+							logger.error("De-activated/deprecated identifier detected for resource identifier {}\nYou should check on identifiers.org to find a new resource identifier for your resource.", resourceIdentifier);
```

Comment:
Is this serious enough to throw an exception or is logging sufficient?

Reply:
:shrug: 
Currently, an error is logged and the accessURL in the reference-databases.xml config file will be used. 

If an exception is thrown, it should probably interrupt the program and cause it to fail. This means that AddLinks will _not_ be able to run if there is an out-of-date resource identifier. A way to get around that is to simply omit the resource identifier (not all databases have resource identifiers in identifiers.org so they are optional). 

We can ask Robin how he wants this handled.

Reply:
Okay - I'm not advocating for either of these.  I just wanted to ask because I didn't know.